client/src/Pages/DashboardProvider/index.js

```diff
@@ -115,7 +163,7 @@ const DashboardProvider = () => {
               <BellFilled />
             </div>
           </div>
-          {page}
+          {isLoading ? <Spin className="UserInfo-icon" /> : page}
```

Comment:
here on the loading case, it will view the spin and all page disappear, you can make a specific component to load on the page not all page